scripts/o3de/o3de/utils.py

```diff
@@ -190,6 +190,15 @@ def download_file(parsed_uri, download_path: pathlib.Path, force_overwrite: bool
                 except KeyError:
                     pass
 
+                def print_progress(downloaded, total_size):
+                    end_ch = '\r'
+                    if (downloaded == total_size):
+                        end_ch = '\n'
+                    print(f'Downloading {object_name} - {downloaded} of {total_size} bytes - {(downloaded/total_size)*100:.2f}%', end=end_ch)
```

Comment:
```suggestion
                    if total_size == 0 or downloaded > total_size:
                        print(f'Downloading {object_name} - {downloaded} bytes', end=end_ch)                     
                    else:
                        if downloaded == total_size:
                            end_ch = '\n'
                        print(f'Downloading {object_name} - {downloaded} of {total_size} bytes - {(downloaded/total_size)*100:.2f}%', end=end_ch)
```
We need to handle the case where `download_file_size` is 0 because `content-length` header is missing.